verifiers/internal/gha/npm_sigstore_tuf_test.go

```
@@ -0,0 +1,120 @@
+		err = json.Unmarshal(content, &expectedKeys)
+		if err != nil {
+			t.Errorf("parsing mock file: %s", err)
+		}
+		mockClient := mockSigstoreTufClient{localPath: testTargetLocalFilePath}
+		actualKeys, err := getNpmjsKeysTarget(mockClient, testTargetLocalFilePath)
+		if err != nil {
+			t.Error(err)
+		}
+		if err != nil {
+			t.Error(err)
+		}
+		if !cmp.Equal(expectedKeys, *actualKeys) {
+			t.Errorf("expected equal values: \nexpected: %v \nactual: %v", expectedKeys, *actualKeys)
+		}
+	})
+
+	t.Run("parsing non-existent registry.npmjs.org_keys.json", func(t *testing.T) {
+		nonExistantPath := "./testdata/my-fake-path"
+		mockClient := mockSigstoreTufClient{localPath: nonExistantPath}
+		_, err := getNpmjsKeysTarget(mockClient, nonExistantPath)
+		if err == nil {
+			t.Error("expected an error")
+		}
+	})
+}
+
+// TestGetKeyDataWithNpmjsKeysTarget ensure that we find the "npm:attestations" key material, given keyid.
+func TestGetKeyDataWithNpmjsKeysTarget(t *testing.T) {
+	tests := []struct {
+		name            string
+		localPath       string
+		keyID           string
+		keyUsage        string
+		expectedKeyData string
+		expectError     bool
+	}{
+		{
+			name:            "npmjs' first attestation key",
+			localPath:       testTargetLocalFilePath,
+			keyID:           testTargetKeyID,
+			keyUsage:        testTargetKeyUsage,
+			expectedKeyData: testTargetKeyData,
+			expectError:     false,
+		},
+		{
+			name:            "missing the 'npm:attestations' keyusage",
+			localPath:       "./testdata/wrong_keyusage_registry.npmjs.org_keys.json",
+			keyID:           testTargetKeyID,
+			keyUsage:        testTargetKeyUsage,
+			expectedKeyData: "", // should not be returned in this error case
+			expectError:     true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockClient := mockSigstoreTufClient{localPath: tt.localPath}
+			keys, err := getNpmjsKeysTarget(mockClient, tt.localPath)
+			if err != nil {
+				t.Error(err)
```

nit: `keys` might be undefined if an error is returned so we should fail early.
```suggestion
				t.Fatalf("getNpmjsKeysTarget: %v", err)
```